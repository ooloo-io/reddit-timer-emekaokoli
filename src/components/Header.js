import React from 'react';
import {
  Container,
  Links,
  LiTag,
  Logo,
  Navlink,
  NavTag,
  // eslint-disable-next-line comma-dangle
  UlTag
} from './common/header/Header.style';

export default function Nav() {
  return (
    <>
      <Container>
        <Navlink to="/">
          <Logo alt={Logo} />
        </Navlink>
        <NavTag>
          <UlTag>
            <LiTag data-testid="search">
              <Navlink to="/search/javascript">Search</Navlink>
            </LiTag>
            <LiTag data-testid="howitworks">
              <Links to="/#how-it-works">How it works</Links>
            </LiTag>
            <LiTag data-testid="about">
              <Links to="/#about">
                About
              </Links>
            </LiTag>
          </UlTag>
        </NavTag>
      </Container>
    </>
  );
}
