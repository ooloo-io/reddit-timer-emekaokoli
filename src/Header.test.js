// import { renderer } from '@testing-library/react';
import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import * as renderer from 'react-test-renderer';
import Header from './components/Header';

describe('Checks header component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Router>
          <Header />
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
