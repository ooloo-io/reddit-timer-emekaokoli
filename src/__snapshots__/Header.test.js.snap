// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checks header component renders correctly 1`] = `
<header
  className="sc-gtssRu ixWDGw"
>
  <a
    aria-current="page"
    className="sc-gKAblj cxXkRZ active"
    href="/"
    onClick={[Function]}
    style={Object {}}
  >
    <img
      alt="elements-logo.svg"
      className="sc-eCApGN kjxHVm"
      src="elements-logo.svg"
    />
  </a>
  <nav
    className="sc-bdnylx lnmxJh"
  >
    <ul
      className="sc-dlnjPT hTAPrA"
    >
      <li
        className="sc-hKFyIo jpiKoV"
        data-testid="search"
      >
        <a
          aria-current={null}
          className="sc-gKAblj cxXkRZ"
          href="/search/javascript"
          onClick={[Function]}
        >
          Search
        </a>
      </li>
      <li
        className="sc-hKFyIo jpiKoV"
        data-testid="howitworks"
      >
        <a
          className="sc-jSFkmK fFLOBj"
          href="/#how-it-works"
          onClick={[Function]}
        >
          How it works
        </a>
      </li>
      <li
        className="sc-hKFyIo jpiKoV"
        data-testid="about"
      >
        <a
          className="sc-jSFkmK fFLOBj"
          href="/#about"
          onClick={[Function]}
        >
          About
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
